gateway/uplink.go

```diff
@@ -66,14 +67,21 @@ func (g *gateway) parseDataUplink(ctx context.Context, phyPayload []byte, packet
 	}
 
 	if downlinkPayload != nil || foptsLength > 0 || sendAck {
-		payload, err := g.createDownlink(device, downlinkPayload, sendAck, fopts)
-		if err != nil {
-			return "", map[string]interface{}{}, fmt.Errorf("failed to create downlink: %w", err)
+		// devices that were configured on older versions of the module will not have these fields and cannot send downlinks.
+		// skip sending downlinks for those devices.
+		if len(device.NwkSKey) == 0 || device.FCntDown == math.MaxUint32 {
+			g.logger.Warnf("Sensor %v must be reset to support new features. "+
```

Comment:
Are we eventually going to sunset this? If so we can add a version number where we say we're no longer going to support the older version.

Reply:
I think once we know the current customers(modern, pret, maybeee eliot) will no longer hit this we should remove it. don't think its worth setting a hard date/version number tho.